scripts/test/miniwdl_validation.py

```diff
@@ -0,0 +1,61 @@
+"""
+This Script checks the provided WDL files with miniwdl for
+syntax, validation, and import errors. See the following
+page on the checks. Note that this script does not
+check for "warning" or "suggestions"; you may use
+`$ miniwdl check WDL_FILENAME` to get warnings and suggestions.
+
+https://miniwdl.readthedocs.io/en/latest/WDL.html#module-WDL.Error
+"""
+
+import argparse
+import sys
+import WDL
+
+
+def format_error_message(error_type, message, pos):
+    return f"{error_type}: '{message}' at {pos.abspath}, Line: {pos.line}, Col: {pos.column}"
+
+
+def check(wdl_filename):
+    try:
+        WDL.load(wdl_filename)
+        return True
+    except WDL.Error.SyntaxError as e:
+        print(format_error_message("Syntax Error", e, e.pos))
+    except WDL.Error.ValidationError as e:
+        print(format_error_message("Validation Error", e, e.node.pos))
+    except WDL.Error.MultipleValidationErrors as e:
+        for ex in e.exceptions:
+            print(format_error_message("Validation Error", ex, ex.node.pos))
+    except WDL.Error.ImportError as e:
+        print(e)
+        print(format_error_message("Import Error", e, e.pos))
+    finally:
+        return False
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("files", nargs="+", help="The WDLs files to be checked.")
+    args = parser.parse_args()
+
+    wdl_filenames = []
+    for filename in args.files:
+        if filename.endswith(".wdl"):
+            wdl_filenames.append(filename)
+    if len(wdl_filenames) != len(args.files):
+        print(f"Warning! Of {len(args.files)} provided files, {len(wdl_filenames)} are WDL.")
```

Comment:
Is there a particular use case you're thinking of with this check? I could see this being useful if a directory were being searched, but since each file is specified I would simply expect validation to fail if one of them isn't a WDL.

Reply:
Would this be a useful check if the script is invoked as `python miniwdl_validation.py wdl/*.*` where the `wdl` directory contains files other than `*.wdl`?

Reply:
I see yes that would be useful, although we usually don't need to validate every wdl, just ones that are part of the "canonical" pipeline i.e. have test inputs available (see validation.sh script for how we do that). Either way, fine to leave it like this.

Reply:
That makes sense. I removed this filtering to simplify the script.